src/sentry/static/sentry/app/utils/withProfiler.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import {Integrations} from '@sentry/apm';
+
+import getDisplayName from 'app/utils/getDisplayName';
+
+type InjectedProps = {
+  finishProfile: () => void;
```

Comment:
I tried to figure out a way to make this type not _required_ to be on the wrapped components type, but not sure if that's possible?

Reply:
I don't think it is possible, as you can't easily inspect the wrapped component's props. Without passing the finishProfile callback this HOC does very little though so I think that `finishProfile` is a reasonable contract to have.